### PR TITLE
Problem: ibc-go 7.3.0 is not used

### PR DESCRIPTION
**Pull request by mmsqe**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

For more [info](https://github.com/cosmos/ibc-go/blob/v7.3.0/docs/migrations/v7_2-to-v7_3.md)
# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,9 +12,9 @@ require (
 	github.com/cometbft/cometbft v0.37.3-0.20230920093934-46df7b597e3c
 	github.com/cometbft/cometbft-db v0.8.0
 	github.com/cosmos/cosmos-proto v1.0.0-beta.2
-	github.com/cosmos/cosmos-sdk v0.47.3
+	github.com/cosmos/cosmos-sdk v0.47.4
 	github.com/cosmos/gogoproto v1.4.10
-	github.com/cosmos/ibc-go/v7 v7.2.0
+	github.com/cosmos/ibc-go/v7 v7.3.0
```
Comment (yihuang) [inline go.mod:17]: `v7.3.1-0.20230920070810-c3261472c815`
the latest `v7.3.x` contains some useful changes

**Review by yihuang — APPROVED** [review]

File: x/cronos/keeper/precompiles/bank.go [hunk 2]
```diff
@@ -134,11 +134,7 @@ func (bc *BankContract) Run(evm *vm.EVM, contract *vm.Contract, readonly bool) (
 		token := args[0].(common.Address)
 		addr := args[1].(common.Address)
 		// query from storage
-		balance := big.NewInt(0)
-		err = stateDB.ExecuteNativeAction(precompileAddr, nil, func(ctx sdk.Context) error {
-			balance = bc.bankKeeper.GetBalance(ctx, sdk.AccAddress(addr.Bytes()), EVMDenom(token)).Amount.BigInt()
-			return err
-		})
+		balance := bc.bankKeeper.GetBalance(stateDB.CacheContext(), sdk.AccAddress(addr.Bytes()), EVMDenom(token)).Amount.BigInt()
```
Comment (calvinaco) [inline x/cronos/keeper/precompiles/bank.go:137]: Is this change supposed to be part of the IBC-go upgrade?

Comment (mmsqe) [reply x/cronos/keeper/precompiles/bank.go:137]: need bump up [ethermint](https://github.com/crypto-org-chain/ethermint/compare/ae144329ecaf...4062d0cc2a32) and bank precompile query api also need

File: x/cronos/keeper/precompiles/interface.go [hunk 3]
```diff
@@ -11,4 +11,5 @@ import (
 type ExtStateDB interface {
 	vm.StateDB
 	ExecuteNativeAction(contract common.Address, converter statedb.EventConverter, action func(ctx sdk.Context) error) error
+	CacheContext() sdk.Context
```
[the remark this answers is not in the document]
Comment (mmsqe) [reply x/cronos/keeper/precompiles/interface.go:14]: yes, do we need separate read only api [change](https://github.com/crypto-org-chain/ethermint/commit/716b0754a135f0ee26a2aa6619769c82aaa108bf) to another PR

File: integration_tests/cosmoscli.py [hunk 4]
```diff
@@ -1315,7 +1315,7 @@ def ica_query_account(self, connid, owner, **kwargs):
             )
         )
 
-    def ica_generate_packet_data(self, tx, memo=None, **kwargs):
+    def ica_generate_packet_data(self, tx, memo=None, encoding="proto3", **kwargs):
```
Comment (yihuang) [inline integration_tests/cosmoscli.py:1318]: shall we default to proto3json in the precompile, will that be more user friendly?

Comment (yihuang) [reply integration_tests/cosmoscli.py:1318]: json encoding need support from counterparty

File: integration_tests/cosmoscli.py [hunk 5]
```diff
@@ -1324,6 +1324,8 @@ def ica_generate_packet_data(self, tx, memo=None, **kwargs):
                 "generate-packet-data",
                 tx,
                 "--memo" if memo else None,
+                "--encoding" if encoding else None,
+                encoding if encoding else None,
```
Comment (yihuang) [inline integration_tests/cosmoscli.py:1328]:
```suggestion
                encoding=encoding,
```

File: integration_tests/cosmoscli.py [hunk 6]
```diff
@@ -1324,6 +1324,8 @@ def ica_generate_packet_data(self, tx, memo=None, **kwargs):
                 "generate-packet-data",
                 tx,
                 "--memo" if memo else None,
```
Comment (yihuang) [inline integration_tests/cosmoscli.py:1326]:
```
memo=memo,
```

File: integration_tests/cosmoscli.py (hunk 4 above)
Comment (mmsqe) [reply integration_tests/cosmoscli.py:1318]: seems `proto3json` doesn't work as expected

Comment (yihuang) [reply integration_tests/cosmoscli.py:1318]: need support from chain-main part I think

**Pull request merged**